src/infrastructure/services/blockstack-service.ts

```diff
@@ -94,4 +110,132 @@ export class BlockstackService {
             return registeredDomainArray[0];
         }
     }
+    public getAddressMap = async (blockstackID: BlockstackId, tag?: string): Promise<IAddressMapping> => {
+        const cruxPayFileName = CruxSpec.blockstack.getCruxPayFilename(blockstackID);
+        return getContentFromGaiaHub(blockstackID.toString(), cruxPayFileName, this.bnsNodes, tag, this.cacheStorage);
+    }
+    public putAddressMap = async (addressMapping: IAddressMapping, cruxDomainId: CruxDomainId, keyManager: IKeyManager): Promise<string> => {
+        const blockstackID = await this.getBlockstackIdFromKeyManager(keyManager, cruxDomainId);
+        if (!blockstackID) {
+            throw errors.ErrorHelper.getPackageError(null, errors.PackageErrorCode.UserDoesNotExist);
+        }
+        const addressDecoder: Decoder<IAddress> = object({
+            addressHash: string(),
+            secIdentifier: optional(string()),
+        });
+        try {
+            for ( const currency of Object.keys(addressMapping) ) {
+                const addressObject: IAddress = addressDecoder.runWithException(addressMapping[currency]);
+            }
+        } catch (error) {
+            throw ErrorHelper.getPackageError(error, PackageErrorCode.AddressMappingDecodingFailure);
+        }
+        const cruxPayFileName = CruxSpec.blockstack.getCruxPayFilename(blockstackID);
+        const gaiaDetails = await getGaiaDataFromBlockstackID(blockstackID.toString(), this.bnsNodes, undefined, this.cacheStorage);
+        const finalURL = await new GaiaService(gaiaDetails.gaiaWriteUrl).uploadContentToGaiaHub(cruxPayFileName, addressMapping, keyManager);
+        log.info(`Address Map for ${blockstackID} saved to: ${finalURL}`);
+        return finalURL;
+    }
+    public getBlockstackIdFromKeyManager = async (keyManager: IKeyManager, cruxDomainId: CruxDomainId): Promise<BlockstackId|undefined> => {
+        const userSubdomainOwnerAddress = publicKeyToAddress(await keyManager.getPubKey());
+        const registeredBlockstackIDs = await this.getRegisteredIDsByAddress(userSubdomainOwnerAddress);
+        console.log(IdTranslator.cruxToBlockstack(cruxDomainId).toString());
+        const registeredDomainArray = registeredBlockstackIDs
+            .map((blockstackID: string) => blockstackID.match(new RegExp(`(.+)\.${IdTranslator.cruxToBlockstack(cruxDomainId).toString()}`)))
+            .map((match) => match && match[0])
+            .filter((domain) => domain !== undefined).filter(Boolean) as string[];
```

Comment:
What does this `.filter(Boolean)` used for?

Reply:
`.filter(Boolean)` makes sure that resultant array does not have non-matching regex as null
Else the resultant array could be `["foo@bar.crux", null, null]`

Reply:
`.filter((domain) => domain !== undefined)` can be removed

Reply:
Removed